src/pages/beneficiaries/Beneficiaries.js

```diff
@@ -12,71 +15,228 @@ const FILTER_MAP = {
 const FILTER_NAMES = Object.keys(FILTER_MAP);
 
 const Beneficiaries = () => {
-    const [beneficiary, setBeneficiary] = useState([]);
+    const [beneficiaries, setBeneficiaries] = useState([]);
+    const [displayedBeneficiaries, setDisplayedBeneficiaries] = useState([]);
     const [search, setSearch] = useState("");
-    const [filter, setFilter] = useState("All");
-
-    // function toggleBfcArchived(id) {
-    //     const updateBfc = beneficiary.map((item) => {
-    //         // if this bfc has the same ID as the edited
-    //         if (id == item.id) {
-    //             return { ...item, archived: !item.archived };
-    //         }
-    //         return item;
-    //     });
-    //     setBeneficiary(updateBfc);
-    // }
-
-    const filterList = FILTER_NAMES.map((name) => (
-        <FilterButton
-            key={name}
-            name={name}
-            isPressed={name === filter}
-            setFilter={setFilter}
-        />
-    ));
+    const [filter, setFilter] = useState("Active");
+    const [activeFilter, setActiveFilter] = useState(true);
+    const [archivedFilter, setArchivedFilter] = useState(false);
+    const [interestsFilter, setInterestsFilter] = useState(false);
+    const [computers, setComputers] = useState(false);
+    const [spokenEnglish, setSpokenEnglish] = useState(false);
+    const [literacy, setLiteracy] = useState(false);
+    const [math, setMath] = useState(false);
+    const [bengali, setBengali] = useState(false);
+    const [arts, setArts] = useState(false);
+    const [bakery, setBakery] = useState(false);
+    const [counseling, setCounseling] = useState(false);
+    const [numInterests, setNumInterests] = useState(0);
```

Comment:
Declaring a piece of boolean state value for each of these seems a little inefficient. Maybe consider using an array and check membership of these values in the array?